File: home_assignments/java/TicketSeller/TicketSellerModel/src/main/java/ticketseller/model/validators/interfaces/IUserValidator.java
package ticketseller.model.validators.interfaces;


import ticketseller.model.entities.User;
import ticketseller.model.validators.ValidationException;

public interface IUserValidator extends IValidator<String, User> {

    /**
     * validates credentials for login functions
     *
     * @param username - said username
     * @param password - said password
     * @throws ValidationException if credentials are null
     */
    void validateForLogin(String username, String password) throws ValidationException;
}

File: home_assignments/csharp/TicketSeller/model/validators/interfaces/IUserValidator.cs
using System;
using model.entities;

namespace model.validators.interfaces;

public interface IUserValidator:IValidator<string,User>
{
    /// <summary>
    /// validates an user for login
    /// </summary>
    /// <param name="username">user's username</param>
    /// <param name="password">user's password</param>
    /// <exception cref="ValidationException">if username or password are null</exception>
    void ValidateForLogin(String username, String password);
}